x-pack/plugins/alerting/server/application/rule/methods/aggregate/types/v1.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { AggregationsAggregationContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { TypeOf } from '@kbn/config-schema';
+import { KueryNode } from '@kbn/es-query';
+import { aggregateOptionsSchema } from '..';
+
+// Adding filter as in schema it's defined as any instead of KueryNode
+export type AggregateOptions = TypeOf<typeof aggregateOptionsSchema> & {
```

Comment:
nice!

A side note: application types and schemas don't really need to be typed, if we typed them we wouldn't be able to use the KueryNode type since that type isn't versioned.

Also a nitpick: I think we should try to be explicit with all of our types so I'd rather we do something like: 

```
type AggregateOptionsSchemaTypes = TypeOf<typeof aggregateOptionsSchema>
type AggregateOptions = {
  search: AggregateOptionsSchemaTypes['search'],
  ...
  filter?: string | KueryNode,
}
```
or whatever variable naming you'd prefer.

Reply:
What's the reason of being explicit with the types? Shouldn't the type always be linked to the schema?